### PR TITLE
Fix check for wrong newlines and renewed check for whitespaces

### PR DESCRIPTION
**Pull request by pascalholthaus**
**What**: Fix check for wrong newlines and renewed check for whitespaces

**Why**:
The test for new lines was broken because python text read changes all newline characters to /n.
The test for spaces was previously misunderstood by me.

**How**:

**Checklist**:

- [ ] Tests
- [ ] Conventional commit message
- [ ] Documentation

### CODE REVIEW

File: tests/plugins/test_files/fail_bad_new_line.nasl [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  script_xref(name:"URL", value:"http://www.foo.org/");
+  script_tag(name:"summary", value:"The remote foo foos the bar");
+
+  script_tag(name:"qod_type", value:"package");
+  script_tag(name:"solution_type", value:"VendorFix");
+
+  exit(0);
+}
aa
```
Comment (Dexus) [inline tests/plugins/test_files/fail_bad_new_line.nasl:52]: is this correct?
```suggestion
}

```

**Review by Dexus — CHANGES_REQUESTED** [review]

File: tests/plugins/test_files/fail_bad_new_line.nasl (hunk 1 above)
Comment (pascalholthaus) [reply tests/plugins/test_files/fail_bad_new_line.nasl:52]: an singel \r is just displayed as "^M"  aa are letters

Comment (pascalholthaus) [reply tests/plugins/test_files/fail_bad_new_line.nasl:52]: Maybe a misunderstanding, it should test extra for \r

File: troubadix/plugins/newlines.py [hunk 2]
```diff
@@ -36,20 +36,25 @@ def run(
         - Search for (\r or \r\n).
         - Search for whitespaces in script_name( "myname") or script_copyright
         """
-
-        if "\r" in file_content:
-            yield LinterError("Found \r.")
+        # Need to be loaded as bytes or \r is converted to \n
+        data = nasl_file.read_bytes().decode(CURRENT_ENCODING)
```
Comment (bjoernricks) [inline troubadix/plugins/newlines.py:40]: I really really thing this is not what you want. The code is now the same as `nasl_files.read_text(encoding=CURRENT_ENCODING)`

Comment (pascalholthaus) [reply troubadix/plugins/newlines.py:40]: if i use this i dont find the /r in fail_bad_new_line.nasl.
Or my unittest for it is broken

Comment (pascalholthaus) [reply troubadix/plugins/newlines.py]: ok changed it to b"/r" in bytes.

**Review by bjoernricks — APPROVED** [review]

**Review by Dexus — APPROVED** [review]

**Pull request merged**